### PR TITLE
ISSUE-545: Added tests for QueryProcessor

### PR DESCRIPTION
**Pull request by ulixius9**
### Describe your changes :

Issue #545 
Added tests for QueryProcessor

#
### Type of change :

- [x] Improvement

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [x] I have performed a self-review of my own. 
- [x] I have tagged my reviewers below.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers

Ingestion: @harshach @ayush-shah @pmbrull

### CODE REVIEW

**Comment by ayush-shah** [comment]
Fixes #545

File: ingestion/tests/unit/test_query_parser.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+from metadata.ingestion.processor.query_parser import QueryParserProcessor
+
+config = """
+{
+  "source": {
+    "type": "sample-usage",
+    "config": {
+      "database": "warehouse",
+      "service_name": "bigquery_gcp",
+      "sample_data_folder": "ingestion/tests/unit/resources"
```
Comment (harshach) [inline ingestion/tests/unit/test_query_parser.py:31]: @ulixius9 @ayush-shah can we add complex queries to the query_log and see if they are passing the test

Comment (ulixius9) [reply ingestion/tests/unit/test_query_parser.py:31]: ok, will add complex queries

**Comment by harshach** [comment]
@ulixius9 can you try adding following queries https://docs.aws.amazon.com/redshift/latest/dg/r_Subquery_examples.html

**Review by harshach — APPROVED** [review]

**Pull request merged**